xonsh/tools.py

```diff
@@ -595,19 +584,16 @@ def ensure_int_or_slice(x):
 
 def is_string_set(x):
     """Tests if something is a set"""
-    if isinstance(x, string_types):
-        return False
-    else:
-        return (isinstance(x, Set) and
-                all([isinstance(a, string_types) for a in x]))
+    return (isinstance(x, Set) and
+            all(isinstance(a, str) for a in x))
 
 
 def csv_to_set(x):
     """Convert a comma-separated list of strings to a set of strings."""
     if not x:
         return set()
     else:
-        return set(x.split(','))
+        return set(y.strip() for y in x.split(','))
```

Comment:
it's used for user-entered data who tend to put an extra space after a comma.

oh, and are sets used here with the rational for uniqueness too? i would porpose another approach to this with the mentioned ensurer enhancement.